src/stacks-hierarchy/types.ts

```diff
@@ -1,36 +1,23 @@
 import {PaymentType} from '@atb/ticketing';
 
-export type CardPaymentMethod =
-  | {paymentType: PaymentType.Visa | PaymentType.Mastercard; save: boolean}
-  | {
-      paymentType: PaymentType.Visa | PaymentType.Mastercard;
-      recurringPaymentId: number;
-    };
-
-export type VippsPaymentMethod = {
-  paymentType: PaymentType.Vipps;
+export type PaymentMethod = {
+  paymentType: PaymentType;
+  recurringPaymentId?: number;
+  save?: boolean;
 };
 
-export type PaymentMethod = VippsPaymentMethod | CardPaymentMethod;
-
 export type SavedRecurringPayment = {
   id: number;
   expires_at: string;
   masked_pan: string;
   payment_type: number;
 };
 
-export type DefaultPaymentOption = {
-  savedType: 'normal';
+export type PaymentOption = {
+  savedType: 'normal' | 'recurring';
   paymentType: PaymentType;
+  recurringCard?: SavedRecurringPayment;
 };
```

Comment:
I'm not completely sold 😅  I prefer verbosity if it gives more precision in the typing. With this change an object like this is possible:
```ts
{
  savedType: 'recurring',
  paymentType: 'Vipps',
  recurringCard: undefined,
}
```

Reply:
I would prefer to remove `savedType` as well, but since it's in local storage, I'm hesitant to touch it. We could maybe mark it as deprecated or something like that.

> I'm not completely sold 

Let me try to sell it, then 😄 

Let's compare it to fare products. What do you think is better?

```ts

type NormalFareProduct = {
  type: 'single' | 'period';
  ...
};
type CarnetFareProduct = {
  id: 'carnet';
  ...
  usedAccesses: Access[];
};
type FareProduct = CarnetFareProduct | NormalFareProduct;
```

or 


```ts
type FareProduct = {
  id: string;
  ...
  usedAccesses: Access[] | undefined;
};
```

One is more data driven. 

- If it has a usedAccesses, we show usedAccesses related UI, otherwise they are all fare products.
- If it has a recurringCard, we show recurringCard related UI, otherwise they are all payment options.

I also appreciate the intention of `SavedPaymentOption = DefaultPaymentOption | RecurringPaymentOption`, but in practice it leads to unnecessary checks for `if ('recurringCard' in paymentOption)` when it could have been a `paymentOption.recurringCard?`.

Reply:
Is a middle ground possible, simply adding `recurringCard: undefined;` to the definition of `DefaultPaymentOption`? So that invalid states remain impossible, but the need for extra `'recurringCard' in paymentOption` checks is removed.

Reply:
Huh, I think that will work! 💡 This gives no typescript errors on my branch:

```ts
type CardPaymentOption = {
  savedType: 'recurring' | 'normal';
  paymentType: PaymentType.Mastercard | PaymentType.Visa;
  recurringCard?: RecurringCard;
};
type VippsPaymentOption = {
  savedType: 'normal';
  paymentType: PaymentType.Vipps;
  recurringCard?: undefined;
};
export type PaymentOption = CardPaymentOption | VippsPaymentOption;
```

Reply:
Nice, we got bag and backpack!